core/expressionParser/helpers/evaluator.js

```diff
@@ -5,37 +5,98 @@ const { isString } = require('@core/stringUtils')
 
 const SystemError = require('@server/utils/systemError')
 
-const unaryEval = async (expr, ctx) => {
+// Built-in functions that can be called, i.e. the standard library.
+// Nothing outside of this set may be used.
+//
+// NB: Namespace conflicts between functions and nodes/variables are allowed
+// I.e. there can be a field called "pow", even if pow(2,3) is a function invocation.
+//
+// stdlib: { [fn]: [Function, min_arity, max_arity? (-1 for infinite)] }
+const stdlib = {
+  pow: [Math.pow, 2], // arity 2
+
+  // arity 1+ (arity 0 allowed by JS)
+  min: [Math.min, 1, -1],
+  max: [Math.max, 1, -1],
+}
+
+const unaryOperators = {
+  '!': a => !a,
+  // TODO: Under JS semantics, "+" coerces a string to a number. Do we want to allow that?
+  '+': a => +a,
+  '-': a => -a,
+}
+
+const binaryOperators = {
+  // Short-circuiting operators (we coerce the output to bool)
+  '||':  (a, b) => !!(a || b),
+  '&&':  (a, b) => !!(a && b),
+  // Normal boolean operators:
+  '==':  (a, b) => a === b,
+  '!=':  (a, b) => a !== b,
+  '===': (a, b) => a === b,
+  '!==': (a, b) => a !== b,
+  '<':   (a, b) => a < b,
+  '>':   (a, b) => a > b,
+  '<=':  (a, b) => a <= b,
+  '>=':  (a, b) => a >= b,
+  // Don't allow bitwise operators:
+  // '|':   (a, b) => a | b,
+  // '^':   (a, b) => a ^ b,
+  // '&':   (a, b) => a & b,
+  // Arithmetic operators:
+  '<<':  (a, b) => a << b,
+  '>>':  (a, b) => a >> b,
+  '>>>': (a, b) => a >>> b,
+  '+':   (a, b) => a + b,
+  '-':   (a, b) => a - b,
+  '*':   (a, b) => a * b,
+  '/':   (a, b) => a / b,
+  '%':   (a, b) => a % b,
+}
+
+const unaryEval = (expr, ctx) => {
   const { argument, operator } = expr
-  const res = await evalExpression(argument, ctx)
-  const x = `${operator} ${JSON.stringify(res)}`
+
+  const fn = unaryOperators[operator]
+  if (!fn) throw new SystemError('undefinedFunction', { fnName: operator })
+
+  const res = evalExpression(argument, ctx)
+
+  // const x = `${operator} ${JSON.stringify(res)}`
   // console.log('=== UNARY')
   // console.log(x)
-  return eval(x)
+  return fn(res)
 }
 
-const binaryEval = async (expr, ctx) => {
+const binaryEval = (expr, ctx) => {
   const { left, right, operator } = expr
-  const leftResult = await evalExpression(left, ctx)
-  const rightResult = await evalExpression(right, ctx)
+
+  const fn = binaryOperators[operator]
+  if (!fn) throw new SystemError('undefinedFunction', { fnName: operator })
+
+  const leftResult = evalExpression(left, ctx)
+  const rightResult = evalExpression(right, ctx)
 
   if (R.isNil(leftResult) || R.isNil(rightResult))
```

Comment:
One thing I wanted to ask about here: If we have e.g. a logical AND here: `null || (foo > 0)`, now it would get evaluated as null. Is this really what we want?

Reply:
Good point @eliask . I think this should be removed. The expression should be evaluated with null values as well. Let's remove the ```if (R.isNil(leftResult) || R.isNil(rightResult))``` check. what do you think?

Reply:
Yeah I'll remove it.

Reply:
Actually it occurred to me that we probably want something that's a bit closer to SQL semantics. So, probably no special handling for logical OR, but return null in every other case.

Reply:
Hm, arguably `+` and `-` could be handled in a different way, would these semantics make sense?

```
1 + null == 1
null + 1 == 1
null - 1 == null
1 - null == 1
```

Reply:
Aand there's a whole another bag of worms if we were to support text operations with `+` :)

Currently there are no type checks so it's possible for booleans, numbers, and strings to get mixed up. And we implicitly therefore support doing various operations with bool+number, number+string, etc.
Maybe no one but me would get the idea to use it, but maybe this could make sense?

```
1 + "text" == "1null" // normal JS semantics
"" + 123 == "123" // normal JS semantics
null + "" == ""
"" + null == ""
"foo" + null + "bar" == "foobar"
```

Reply:
For now I would leave as normal JS semantics... we'll update the behavior in the future if needed. what do you think?